src/extension.ts

```diff
@@ -161,6 +161,12 @@ export function activate(context: vscode.ExtensionContext): void {
             node = target;
         }
 
+        let newApp: boolean = false;
+        // if the tree changes here, a new app was created
+        tree.onDidChangeTreeData(() => {
```

Comment:
This is an interesting approach, but I'm worried about timing issues. For example, what if I want to deploy to "WebApp1" which happens to be stopped, and I do the following:
1. Right click WebApp1 -> Start
1. Select Deploy
1. Select Subscription
&lt;WebApp1 is started and tree refreshes so that the label is updated&gt;
1. Select WebApp1

Expected:
I'm prompted to confirm deployment since WebApp1 already exists

Actual:
I'm not prompted because the label refresh triggered onDidChangeTreeData

Reply:
Yeah, you're right.  I had trouble reproducing with those steps (since stop => start happens pretty quickly on my machine) but if I do this:

1.) Stop WebApp 1
2.) Start a deploy
3.) Before I pick a node, start WebApp1
4.) Deploy to WebApp1

I was thinking alternatively I could check the number of web apps before and after the node select action.  Because creating a new web app opens up a new wizard (and closes the deploy wizard), we shouldn't run into the same issue.

Thoughts?

Reply:
I would advocate for changing something inside the shared package. I feel like you're hesitant to change it, but I think that's the only concrete way to make sure we don't have timing issues. Even with your alternative above, it would be complicated if I create multiple web apps at the same time.

Here are a couple ideas off the top of my head:
1. Change the return value of showNodePicker from Promise<IAzureNode> to Promise<[IAzureNode, boolean]> (easy to explain, but annoying to implement because its a breaking change)
1. Add another event to AzureTreeDataProvider - something like 'onNodeCreated' that passes the node or node's id to the listener. Then you can check if that matches the node returned by showNodePicker

Reply:
Just saw this:
>  Because creating a new web app opens up a new wizard (and closes the deploy wizard), we shouldn't run into the same issue.

But there's still ways we could run into issues. For example I create a new web app while deploying, but also delete a web app during that time.

Reply:
Closing in lieu of option 2.